pkg/apiserver/webhook/configmap_webhook.go

```diff
@@ -106,24 +110,26 @@ func (v *ConfigMapValidator) validateUpdate(req admission.Request) admission.Res
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	oldCloudResourcePrefix, err := v.getCloudResourcePrefix(oldConfigMap)
+	oldControllerConfig, err := v.getControllerConfig(oldConfigMap)
 	if err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	if newCloudResourcePrefix == oldCloudResourcePrefix {
-		return admission.Allowed("")
-	}
-	if v.NpControllerInterface.IsCloudResourceCreated() {
-		return admission.Denied(fmt.Sprintf("cloud resource is already created with CloudResourcePrefix %s. "+
-			"Please delete the resource and try or use the same CloudResourcePrefix",
-			oldCloudResourcePrefix))
-	}
-	if len(newCloudResourcePrefix) > 0 {
-		if !config.ValidateName(newCloudResourcePrefix) {
+	if newControllerConfig.CloudResourcePrefix != oldControllerConfig.CloudResourcePrefix {
+		if v.NpControllerInterface.IsCloudResourceCreated() {
+			return admission.Denied(fmt.Sprintf("cloud resource is already created with CloudResourcePrefix %s. "+
+				"Please delete the resource(s) and retry", oldControllerConfig.CloudResourcePrefix))
+		}
+		if !config.ValidateName(newControllerConfig.CloudResourcePrefix) {
```

Comment:
shouldnt validate happens before checking for cloud resources ?

Reply:
We can but this approach we can avoid validates the name if cloud resource is already created.